android/app/src/main/kotlin/org/getlantern/lantern/model/InAppBilling.kt

```diff
@@ -318,4 +331,36 @@ class InAppBilling(
             }
         }
     }
+
+    private fun addAcknowledgePurchase(planID: String, currency: String, token: String) {
+        val session = LanternApp.getSession()
+        val json = JsonObject()
+        json.addProperty("idempotencyKey", System.currentTimeMillis().toString())
+        json.addProperty("provider", PaymentProvider.GooglePlay.toString().lowercase())
+        json.addProperty("email", session.email())
+        json.addProperty("plan", planID)
+        json.addProperty("currency", currency.lowercase())
+        json.addProperty("deviceName", session.deviceName())
+        json.addProperty("token", token)
+
+        lanternClient.post(
+            LanternHttpClient.createProUrl("/purchase"),
```

Comment:
I'm a little unclear on why we're adding this. Aren't we already calling the purchase endpoint in `sendPurchaseRequest` in `PaymentUtil.kt`?

Reply:
the reason why we are adding here, if purchase takes some time process then,if server try to make acknowledgment and purchase had not been process then it will throw error, so when user restarts app, will check for any purchases that has not been acknowledged, if found we need to let server know that purchase state has changed and make acknowledgment.